packages/synapse-react-client/src/components/EntityViewScopeEditor/EntityViewScopeEditorModal.test.tsx

```diff
@@ -0,0 +1,235 @@
+import { createWrapper } from '../../testutils/TestingLibraryUtils'
+import React from 'react'
+import EntityViewScopeEditorModal from './EntityViewScopeEditorModal'
+import EntityViewScopeEditor from './EntityViewScopeEditor'
+import { EntityViewScopeEditorModalProps } from './EntityViewScopeEditorModal'
+import { act, render, screen, within } from '@testing-library/react'
+import { waitFor } from '@testing-library/react'
+import SynapseClient from '../../synapse-client'
+import userEvent from '@testing-library/user-event'
+import mockTableEntityData, {
+  mockTableEntity,
+} from '../../mocks/entity/mockTableEntity'
+import {
+  ENTITY_VIEW_TYPE_MASK_FILE,
+  ENTITY_VIEW_TYPE_MASK_FOLDER,
+  ENTITY_VIEW_TYPE_MASK_PROJECT,
+  ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+  Entity,
+} from '@sage-bionetworks/synapse-types'
+import { MOCK_ACCESS_TOKEN } from '../../mocks/MockSynapseContext'
+import { SynapseClientError } from '../../utils'
+import { mockProjectViewEntity } from '../../mocks/entity/mockProjectView'
+
+const mockTableEntityInstance = mockTableEntityData.entity
+
+jest.mock('../EntityViewScopeEditor/EntityViewScopeEditor', () => ({
+  __esModule: true,
+  default: jest.fn(() => <div data-testid={'EntityViewScopeEditorrMocked'} />),
+}))
+const mockEntityViewScopeEditor = jest.mocked(EntityViewScopeEditor)
+
+function renderComponent(props: EntityViewScopeEditorModalProps) {
+  return render(<EntityViewScopeEditorModal {...props} />, {
+    wrapper: createWrapper(),
+  })
+}
+
+const mockUpdateEntity = jest.spyOn(SynapseClient, 'updateEntity')
+
+describe('EntityViewScopeEditorModal tests', () => {
+  const mockOnCancel = jest.fn()
+  const mockOnUpdate = jest.fn()
+
+  beforeEach(() => {
+    jest.spyOn(SynapseClient, 'getEntity').mockResolvedValue({
+      ...mockTableEntity,
+      scopeIds: ['syn123'],
+      viewTypeMask: ENTITY_VIEW_TYPE_MASK_FILE | ENTITY_VIEW_TYPE_MASK_FOLDER,
+      concreteType: ENTITY_VIEW_CONCRETE_TYPE_VALUE,
+    } as Entity)
+  })
+
+  it('displays the correct scope editor and mask editor after initial render', async () => {
+    jest.clearAllMocks()
+    renderComponent({
+      entityId: mockTableEntity.id,
+      open: true,
+      onCancel: mockOnCancel,
+      onUpdate: mockOnUpdate,
+    })
+
+    const dialog = screen.getByRole('dialog')
+    expect(
+      within(dialog).getByRole('heading', { name: 'Edit Scope' }),
+    ).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Cancel' })).toBeVisible()
+    expect(within(dialog).getByRole('button', { name: 'Save' })).toBeVisible()
+
+    const filesButton = within(dialog).getByRole('checkbox', { name: 'Files' })
```

Comment:
Prefer renaming these `filesCheckbox` or something like that; these are not buttons